connector/apmconnector/metric_to_metric_connector.go

```diff
@@ -0,0 +1,257 @@
+
+		for j := 0; j < rm.ScopeMetrics().Len(); j++ {
+			sm := rm.ScopeMetrics().At(j)
+
+			smNew := pmetric.ScopeMetrics{}
+
+			for k := 0; k < sm.Metrics().Len(); k++ {
+				m := sm.Metrics().At(k)
+
+				if m.Name() == "http.server.duration" || m.Name() == "http.server.request.duration" {
+					rmNew, smNew, metrics = createResourceAndScopeMetrics(logger, rmNew, attributesFilter, rm, newMetrics, metrics, metricMap, smNew)
+					recordTransactionMetrics(logger, m, metrics, apdex, smNew)
+				} else if m.Name() == "http.client.duration" || m.Name() == "http.client.request.duration" {
+					rmNew, smNew, metrics = createResourceAndScopeMetrics(logger, rmNew, attributesFilter, rm, newMetrics, metrics, metricMap, smNew)
+					recordExternalHostDurationMetric(logger, m, smNew)
+				}
+			}
+		}
+	}
+
+	if newMetrics.ResourceMetrics().Len() > 0 {
+		metricMap.AppendOtelMetrics(newMetrics)
+	}
+
+	return newMetrics
+}
+
+func recordExternalHostDurationMetric(logger *zap.Logger, m pmetric.Metric, smNew pmetric.ScopeMetrics) {
+	newMetric := pmetric.NewMetric()
+	newMetric.SetName("apm.service.external.host.duration")
+	newMetric.SetDescription("Duration of external calls")
+	newMetric.SetUnit(m.Unit())
+
+	switch metricType := m.Type(); metricType {
+	case pmetric.MetricTypeHistogram:
+		newMetric.SetEmptyHistogram().DataPoints().EnsureCapacity(3)
+		for i := 0; i < m.Histogram().DataPoints().Len(); i++ {
+			dp := m.Histogram().DataPoints().At(i)
+			if serverAddress, hasServerAddress := GetServerAddress(dp.Attributes()); hasServerAddress {
+				newDp := newMetric.Histogram().DataPoints().AppendEmpty()
+				dp.CopyTo(newDp)
+				newDp.Attributes().PutStr("server.address", serverAddress)
+				newDp.Attributes().PutStr("external.host", serverAddress)
+				newDp.Attributes().PutStr("metricTimesliceName", fmt.Sprintf("External/%s/all", serverAddress))
+			}
+		}
+		newMetric.Histogram().SetAggregationTemporality(m.Histogram().AggregationTemporality())
+		newMetric.CopyTo(smNew.Metrics().AppendEmpty())
+	case pmetric.MetricTypeExponentialHistogram:
+		newMetric.SetEmptyExponentialHistogram().DataPoints().EnsureCapacity(3)
+		for i := 0; i < m.ExponentialHistogram().DataPoints().Len(); i++ {
+			dp := m.ExponentialHistogram().DataPoints().At(i)
+			if serverAddress, hasServerAddress := GetServerAddress(dp.Attributes()); hasServerAddress {
+				newDp := newMetric.ExponentialHistogram().DataPoints().AppendEmpty()
+				dp.CopyTo(newDp)
+				newDp.Attributes().PutStr("server.address", serverAddress)
+				newDp.Attributes().PutStr("external.host", serverAddress)
+				newDp.Attributes().PutStr("metricTimesliceName", fmt.Sprintf("External/%s/all", serverAddress))
+			}
+		}
+		newMetric.ExponentialHistogram().SetAggregationTemporality(m.ExponentialHistogram().AggregationTemporality())
+		newMetric.CopyTo(smNew.Metrics().AppendEmpty())
+	default:
+		logger.Error("unexpected metric type", zap.String("name", m.Name()), zap.String("type", metricType.String()))
```

Comment:
maybe add a comment pointing the the semantic conventions that specify this metric should be a histogram